internal/tester.go

```diff
@@ -34,35 +45,93 @@ type tester struct {
 }
 
 func (t *tester) executeTests() error {
-	if err := t.writeKuttlFiles(); err != nil {
-		return errors.Wrap(err, "cannot write kuttl test files")
+	if err := writeTestFile(t.manifests, t.options.Directory); err != nil {
+		return err
+	}
+
+	resources, err := t.writeChainsawFiles()
+	if err != nil {
+		return errors.Wrap(err, "cannot write chainsaw test files")
+	}
+
+	fmt.Printf("Written test files: %s\n", t.options.Directory)
+
+	if t.options.RenderOnly {
+		return nil
 	}
-	fmt.Println("Running kuttl tests at " + t.options.Directory)
-	cmd := exec.Command("bash", "-c", fmt.Sprintf(`"${KUTTL}" test --start-kind=false --skip-cluster-delete %s --timeout %d 2>&1`, t.options.Directory, t.options.DefaultTimeout)) // #nosec G204
-	stdout, _ := cmd.StdoutPipe()
-	if err := cmd.Start(); err != nil {
-		return errors.Wrap(err, "cannot start kuttl")
+
+	fmt.Println("Running chainsaw tests at " + t.options.Directory)
+	for _, tf := range testFiles {
+		if !checkFileExists(filepath.Join(t.options.Directory, caseDirectory, tf)) {
+			continue
+		}
+		cmd := exec.Command("bash", "-c", fmt.Sprintf(`"${CHAINSAW}" test --test-dir %s --test-file %s --skip-delete --parallel 1 2>&1`, filepath.Join(t.options.Directory, caseDirectory), tf)) // #nosec G204
+		stdout, _ := cmd.StdoutPipe()
+		if err := cmd.Start(); err != nil {
+			return errors.Wrap(err, "cannot start chainsaw")
+		}
+
+		var mutex sync.Mutex
+		// Start ticker for kubectl command every 30 seconds
+		ticker := time.NewTicker(t.options.LogCollectionInterval)
+		done := make(chan bool)
+		go logCollector(done, ticker, &mutex, resources)()
+
+		sc := bufio.NewScanner(stdout)
+		sc.Split(bufio.ScanLines)
+		for sc.Scan() {
+			mutex.Lock()
+			fmt.Println(sc.Text())
+			mutex.Unlock()
+		}
+		if err := cmd.Wait(); err != nil {
+			return errors.Wrap(err, "cannot wait for chainsaw")
+		}
+
+		ticker.Stop()
+		done <- true
 	}
-	sc := bufio.NewScanner(stdout)
-	sc.Split(bufio.ScanLines)
-	for sc.Scan() {
-		fmt.Println(sc.Text())
+	return nil
+}
+
+func logCollector(done chan bool, ticker *time.Ticker, mutex sync.Locker, resources []config.Resource) func() {
+	return func() {
+		for {
+			select {
+			case <-done:
+				return
+			case <-ticker.C:
+				mutex.Lock()
+				fmt.Printf("crossplane trace logs %s\n", time.Now())
+				for _, r := range resources {
+					kubectlCmd := exec.Command("bash", "-c", fmt.Sprintf(`"${CROSSPLANE_CLI}" beta trace %s %s -o wide`, r.KindGroup, r.Name)) //nolint:gosec // Disabling gosec to allow dynamic shell command execution
+					output, err := kubectlCmd.CombinedOutput()
+					if err != nil {
+						fmt.Println("Error executing crossplane:", err)
```

Comment:
The idea here is, not exiting because, during deletion sometimes this command may fail for some resources. In this case, we want to continue with other resources.